node/router/src/inbound.rs

```diff
@@ -341,25 +341,25 @@ pub trait Inbound<N: Network>: Reading + Outbound<N> {
 
         // Truncate and convert to socket addrs.
         peers.truncate(MAX_PEERS_TO_SEND);
-        let peers = peers.into_iter().map(|peer| peer.listener_addr).collect();
+        let peers = peers.into_iter().map(|peer| (peer.listener_addr, peer.last_height_seen)).collect();
 
         // Send a `PeerResponse` message to the peer.
         self.router().send(peer_ip, Message::PeerResponse(PeerResponse { peers }));
         true
     }
 
     /// Handles a `PeerResponse` message.
-    fn peer_response(&self, _peer_ip: SocketAddr, peers: &[SocketAddr]) -> bool {
+    fn peer_response(&self, _peer_ip: SocketAddr, peers: &[(SocketAddr, Option<u32>)]) -> bool {
         // Check if the number of peers received is less than MAX_PEERS_TO_SEND.
         if peers.len() > MAX_PEERS_TO_SEND {
             return false;
         }
-        // Filter out invalid addresses.
+        // Filter out invalid addresses and peers with a lower known block height.
```

Comment:
Where does the filtering of lower block height happen? I might be missing something.

Reply:
No, you're right, it must have been some earlier change that got superseded.